test/hummingbot/client/command/test_balance_command.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+import unittest
+from typing import Awaitable
+from unittest.mock import patch, MagicMock
+
+from hummingbot.client.config.global_config_map import global_config_map
+from hummingbot.client.hummingbot_application import HummingbotApplication
+from test.mock.mock_cli import CLIMockingAssistant
+
+
+class BalanceCommandTest(unittest.TestCase):
+    @patch("hummingbot.core.utils.trading_pair_fetcher.TradingPairFetcher")
+    def setUp(self, _: MagicMock) -> None:
+        super().setUp()
+        self.ev_loop = asyncio.get_event_loop()
+        self.app = HummingbotApplication()
+        self.cli_mock_assistant = CLIMockingAssistant()
+        self.cli_mock_assistant.start()
+
+    def tearDown(self) -> None:
+        super().tearDown()
+        self.cli_mock_assistant.stop()
+
+    @staticmethod
+    async def raise_timeout(*args, **kwargs):
+        raise asyncio.TimeoutError
+
+    def async_run_with_timeout(self, coroutine: Awaitable, timeout: float = 1):
+        ret = self.ev_loop.run_until_complete(asyncio.wait_for(coroutine, timeout))
+        return ret
+
+    def async_run_with_timeout_coroutine_must_raise_timeout(self, coroutine: Awaitable, timeout: float = 1):
+        class DesiredError(Exception):
+            pass
+
+        async def run_coro_that_raises(coro: Awaitable):
+            try:
+                await coro
+            except asyncio.TimeoutError:
+                raise DesiredError
```

Comment:
Care to explain the necessity of creating a `DesiredError` here?

Reply:
Yes, it's because we want to test that the function raises an `asyncio.TimeoutError`, but we also run the functions in an `asyncio.wait_for` which also raises that error. This means that, if a function gets stuck at some point during its execution, `asyncio.wait_for` will raise `asyncio.TimeoutError` but the test case will wrongly pass. So, it's basically to differentiate between the timeout we want and the one we don't.